### PR TITLE
Update the changelog (pre-0.7.0)

### PR DESCRIPTION
**Pull request by aldanor**

### CODE REVIEW

**Comment by aldanor** [comment]
@mulimoen Mostly your work here, so cheers for that :)

Really hope docs.rs build will work out, but I guess we'll know soon enough.

Also hoping to get this merged asap so I get back to finishing up the mega branch I've had stashed for almost a year now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+  HDF5 installation methods covered: conda, apt, homebrew, also official
+  binaries on Windows. We now also test MPI versions of the library for
+  both MPICH and OpenMPI on macOS / Linux.
+
+### Fixed
+
+- We now force the variable-length allocator that HDF5 uses when reading data
+  to use `libc::malloc` and `libc::free`, so that they can be deallocated
+  properly by `VarLenString` and `VarLenArray` in `hdf5-types`. Previously,
+  this could cause a rare but serious failure when the default allocator
```
Comment (mulimoen) [inline CHANGELOG.md:30]: on windows

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,39 @@
 # Changelog
 
+## 0.7.0
+
+### Added
+
+- HDF5 C library can now be built from source and linked in statically, enabled
+  via `hdf5-sys/static` feature (as of this release, library version is 1.10.5).
```
Comment (mulimoen) [inline CHANGELOG.md:8]: 1.10.6

Comment (aldanor) [reply CHANGELOG.md]: Good catch, fixe

File: CHANGELOG.md (hunk 1 above)
Comment (aldanor) [reply CHANGELOG.md]: Yep, added that

**Comment by aldanor** [comment]
Updated; for some reason, static builds were listed on 0.6.1, fixed that too.

**Pull request merged**